sympy/solvers/solvers.py

```diff
@@ -1018,11 +1018,10 @@ def _sympified_list(w):
             ok = True
         else:
             if fi.is_number:
-                if fi.is_Number:
-                    if fi.is_zero:
-                        continue
+                if fi.is_zero is False:
                     return []
-                ok = True
+                if not fi.is_Number is True:
+                    ok = True
```

Comment:
I find this change quite hard to parse. The reason for using `is False` with `fi.is_zero` is because it has three possible values (True, False and None). That isn't the case for `is_Number` which is always True or False.

Otherwise I'm not sure I follow what the changed logic here is doing. Previously this would `continue` if `fi.is_zero` is true. It doesn't look like it still does that.

Reply:
> Previously this would `continue` if fi.is_zero is true. It doesn't look like it still does that.

I think the keyword you're mentioning is `pass`. Instead `continue` keyword will skip the rest of the code inside a loop for the current iteration. 

https://github.com/sympy/sympy/blob/17cf0f5ad8c2535e32009869220654e7fd25865f/sympy/solvers/solvers.py#L1016-L1034

Which means previously if the expression `is_Number` and `is_zero` then the loop will continue onto it's next iteration without the `ok = True` (#L1025) or the `newf.append(fi)` (#L1030) statements being executed for the current iteration. (This might be because `is_zero` would be either `True` or `False` for are `Number` instances). This is as same as the current logic where the expression is being ignored if it `is_Number` and `is_zero`.

Reply:
Can we just rewrite the whole loop as:
```python
if fuzzy_or(fuzzy_not(fi.is_zero) for fi in f):
    return []

def ok(fi):
    if fi.free_symbols & symset:
        return True
    else:
        return fi.is_number or fi.is_constant()

f = [fi for fi in f if ok(fi)]
```
I guess that the intention is to try and avoid checking `is_zero` unless absolutely necessary but it could be simpler like:
```python
for fi in f:
    if fi.free_symbols & symset:
        newf.append(fi)
    elif fi.is_number:
        if fi.is_zero is False:
            return []
        else:
            newf.append(fi)
    elif fi.is_constant():
        newf.append(fi)
    else:
        # Skip this equation
        continue
```

Reply:
I tried implementing your suggested loop but it still seems to be failing tests. It seems to me that there are instances of `Number` such as `Matrix` (one of the failing tests ) which are not at all meant to be passed in `Poly` class. Hence initially (even before this PR) the logic was such that it handled at least all of the `Number` instances in this loop itself rather than letting them pass. This seems to be different for the expressions which are `number` but not `Number`, and they can be given forward for further processing. The current issue for this PR seems to have arisen from the same whence a `sqrt` which was supposed to be a `Number` but isn't was passed on forward into `Poly` class.

Reply:
I don't think that we should use the loop here as a way of preventing an expression from being passed to `Poly` later. This loop here is to try and handle equations not involving the unknowns based on whether or not the are known to be zero and that is the only role it should play.

The problem with `Poly(sqrt(2))` raising should be handled at the place where `Poly` is called.

Reply:
You're right there are too many variables here we can just catch this error in as_poly(), similar to how PolynomialError is being caught there. 

This'll also handle the cases when user tries to create objects like `Poly(sqrt(2))` seperately.